controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -652,6 +666,134 @@ bool master_thread::handle_cmdu_1905_channel_preference_report_message(
+            return false;
+        }
+
+        if (tlvType != int(wfa_map::eTlvTypeMap::TLV_CHANNEL_SELECTION_RESPONSE)) {
+            LOG(ERROR) << "Unexpected tlv type in TLV_CHANNEL_SELECTION_RESPONSE, type="
+                       << int(tlvType);
+            // TODO: replace return statement with function that skip s unexpected tlv
+            // see: https://github.com/prplfoundation/prplMesh/issues/107
+            return false;
+        }
+
+        auto channel_selection_response_tlv =
+            cmdu_rx.addClass<wfa_map::tlvChannelSelectionResponse>();
+        if (!channel_selection_response_tlv) {
+            LOG(ERROR) << "addClass ieee1905_1::tlvChannelSelectionResponse has failed";
+            return false;
+        }
+
+        auto &ruid         = channel_selection_response_tlv->radio_uid();
+        auto response_code = channel_selection_response_tlv->response_code();
+
+        LOG(DEBUG)
+            << "channel selection response from ruid=" << network_utils::mac_to_string(ruid)
+            << ", response_code="
+            << ([](const wfa_map::tlvChannelSelectionResponse::eResponseCode &response_code) {
+                   std::string ret_str;
+                   switch (response_code) {
+                   case wfa_map::tlvChannelSelectionResponse::eResponseCode::ACCEPT:
+                       ret_str.assign("ACCEPT");
+                       break;
+                   case wfa_map::tlvChannelSelectionResponse::eResponseCode::
+                       DECLINE_VIOLATES_CURRENT_PREFERENCES:
+                       ret_str.assign("DECLINE_VIOLATES_CURRENT_PREFERENCES");
+                       break;
+                   case wfa_map::tlvChannelSelectionResponse::eResponseCode::
+                       DECLINE_VIOLATES_MOST_RECENTLY_REPORTED_PREFERENCES:
+                       ret_str.assign("DECLINE_VIOLATES_MOST_RECENTLY_REPORTED_PREFERENCES");
+                       break;
+                   case wfa_map::tlvChannelSelectionResponse::eResponseCode::
+                       DECLINE_PREVENT_OPERATION_OF_BACKHAUL_LINK:
+                       ret_str.assign("DECLINE_PREVENT_OPERATION_OF_BACKHAUL_LINK");
+                       break;
+                   default:
+                       ret_str.assign("ERROR:UNFAMILIAR_RESPONSE_CODE");
+                       break;
+                   }
+                   return ret_str;
+               })(response_code);
+
+    } //close while (cmdu_rx.getNextTlvType(tlvType))
+
+    return true;
+}
+
+bool master_thread::handle_cmdu_1905_operating_channel_report(Socket *sd,
+                                                              ieee1905_1::CmduMessageRx &cmdu_rx)
+{
+    auto mid = cmdu_rx.getMessageId();
+    LOG(INFO) << "Received OPERATING_CHANNEL_REPORT_MESSAGE, mid=" << std::dec << int(mid);
+
+    int tlvType;
+    while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE)) {
```

Comment:
The check for tlvType is redundant - in any case you can't support TLVs which are out of order, since we don't support skip TLV in TLVF. So you expect TLV type to be `TLV_OPERATING_CHANNEL_REPORT` anyway so just do `addClass` which has to succeed - if it doesn't, its a failure.
Then all that's left is to stop on `TLV_END_OF_MESSAGE` - so the best loop structure for that is do - while:
```
do {
  auto operating_channel_report_tlv = cmdu_rx.addClass<wfa_map::tlvOperatingChannelReport>();
  if (!operating_channel_report_tlv) {
    LOG(ERROR) << "addClass ieee1905_1::operating_channel_report_tlv has failed";
    return false;
  }
...
...
} while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE));
```

Reply:
Please do the same in the `son_slave_thread` as part of this PR, it will make the function much simpler and readable.

Reply:
> The check for tlvType is redundant - in any case you can't support TLVs which are out of order, since we don't support skip TLV in TLVF. So you expect TLV type to be `TLV_OPERATING_CHANNEL_REPORT` anyway so just do `addClass` which has to succeed - if it doesn't, its a failure.
> Then all that's left is to stop on `TLV_END_OF_MESSAGE` - so the best loop structure for that is do - while:
> 
> ```
> do {
>   auto operating_channel_report_tlv = cmdu_rx.addClass<wfa_map::tlvOperatingChannelReport>();
>   if (!operating_channel_report_tlv) {
>     LOG(ERROR) << "addClass ieee1905_1::operating_channel_report_tlv has failed";
>     return false;
>   }
> ...
> ...
> } while ((tlvType = cmdu_rx.getNextTlvType()) != int(ieee1905_1::eTlvType::TLV_END_OF_MESSAGE));
> ```



> Please do the same in the `son_slave_thread` as part of this PR, it will make the function much simpler and readable.

This can be done if expecting to only one type of TLV. Wherever several TLV types can be received, the order of the TLVs is not defined in the standard.

I changed it to do-while on handlers of channel selection response and operating channel report handlers.